pkg/queue/manager.go

```diff
@@ -59,25 +123,24 @@ func (m *Manager) AddQueue(ctx context.Context, q *kueue.Queue) error {
 	if _, ok := m.queues[key]; ok {
 		return fmt.Errorf("queue %q already exists", q.Name)
 	}
-	qImpl := newQueue()
-	qImpl.setProperties(q)
+	qImpl := newQueue(q)
 	m.queues[key] = qImpl
 	// Iterate through existing workloads, as workloads corresponding to this
 	// queue might have been added earlier.
 	var workloads kueue.QueuedWorkloadList
 	if err := m.client.List(ctx, &workloads, client.MatchingFields{workloadQueueKey: q.Name}, client.InNamespace(q.Namespace)); err != nil {
 		return fmt.Errorf("listing workloads that match the queue: %w", err)
 	}
-	addedWorkloads := false
-	for i, w := range workloads.Items {
+	for _, w := range workloads.Items {
+		w := w
 		// Checking queue name again because the field index is not available in tests.
 		if w.Spec.QueueName != q.Name || w.Spec.Admission != nil {
 			continue
 		}
-		qImpl.PushOrUpdate(&workloads.Items[i])
-		addedWorkloads = true
+		qImpl.items[workload.Key(&w)] = &w
 	}
-	if addedWorkloads {
+	cq := m.clusterQueues[qImpl.ClusterQueue]
```

Comment:
How about if the ClusterQueue doesn't exist?

Reply:
Nothing happens, those workloads are ignored. When the ClusterQueue is created, we insert all the workloads from queues that match.

Reply:
we need to find a way to update the status condition of those workloads to explain why they are pending.

Reply:
created #102